### DISCUSSION TITLE
math: integer division, modulo?

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#1064] [Q&A] [answered]
_Originally opened by @vikstrous2 in https://github.com/cuelang/cue/discussions/1064_

Hey guys, I'm struggling with doing just normal integer stuff in a for comprehension. I want to loop through a list and set x to idx%! (MISSING)and y to idx/2 using integer division (rounding down). However, I'm fighting the type system and I can't find a good way to do it. This is the ideal way I would hope to achieve what I want:

```
[ for idx, endpoint in _endpoints {
x: idx%!
(MISSING)
y: idx/2
}]
```

Since idx is int, not float64, I can't use it in any of the math library functions, like Mod. However, if I divide it at all, it goes from an int to a float (still not float64). I can't do integer division and I can't do modulo and I keep feeling like I must be missing something obvious.

**Comment by cueckoo** [comment]
_Original reply by @vikstrous2 in https://github.com/cuelang/cue/discussions/1064#discussioncomment-884195_

Ok, yes, I was missing something obvious. Right after posting, I found a good way to do what I want:
```
x: idx-(math.Floor(idx/2)*2)
y: math.Floor(idx/2)
```
This solves my specific use case and hopefully it helps others, but working with numbers still feels a bit annoying.

**Comment by myitcv** [comment, marked as answer]
@vikstrous2 apologies for the delay in replying. I think you're after:

https://cuelang.org/docs/references/spec/#div-mod-quo-and-rem

**Reply by vikstrous2** [reply]
Yep. It would be great to mark this as the right answer, but now the bot owns the discussion.

**Reply by myitcv** [reply]
Done!
